nixos/tests/k3s/multi-node.nix

```diff
@@ -183,52 +179,42 @@ import ../make-test-python.nix (
         };
     };
 
-    testScript = ''
-      machines = [server, server2, agent]
-      for m in machines:
-          m.start()
-          m.wait_for_unit("k3s")
-
-      is_aarch64 = "${toString pkgs.stdenv.hostPlatform.isAarch64}" == "1"
-
-      # wait for the agent to show up
-      server.wait_until_succeeds("k3s kubectl get node agent")
-
-      for m in machines:
-          m.succeed("k3s check-config")
-          m.succeed(
-              "${pauseImage} | k3s ctr image import -"
-          )
-
-      server.succeed("k3s kubectl cluster-info")
-      # Also wait for our service account to show up; it takes a sec
-      server.wait_until_succeeds("k3s kubectl get serviceaccount default")
-
-      # Now create a pod on each node via a daemonset and verify they can talk to each other.
-      server.succeed("k3s kubectl apply -f ${networkTestDaemonset}")
-      server.wait_until_succeeds(f'[ "$(k3s kubectl get ds test -o json | jq .status.numberReady)" -eq {len(machines)} ]')
-
-      # Get pod IPs
-      pods = server.succeed("k3s kubectl get po -o json | jq '.items[].metadata.name' -r").splitlines()
-      pod_ips = [server.succeed(f"k3s kubectl get po {name} -o json | jq '.status.podIP' -cr").strip() for name in pods]
-
-      # Verify each server can ping each pod ip
-      for pod_ip in pod_ips:
-          server.succeed(f"ping -c 1 {pod_ip}")
-          agent.succeed(f"ping -c 1 {pod_ip}")
-
-      # Verify the pods can talk to each other
-      resp = server.wait_until_succeeds(f"k3s kubectl exec {pods[0]} -- socat TCP:{pod_ips[1]}:8000 -")
-      assert resp.strip() == "server"
-      resp = server.wait_until_succeeds(f"k3s kubectl exec {pods[1]} -- socat TCP:{pod_ips[0]}:8000 -")
-      assert resp.strip() == "server"
-
-      # Cleanup
-      server.succeed("k3s kubectl delete -f ${networkTestDaemonset}")
-
-      for m in machines:
-          m.shutdown()
-    '';
+    testScript = # python
+      ''
+        start_all()
+
+        machines = [server, server2, agent]
+        for m in machines:
+            m.wait_for_unit("k3s")
+
+        # wait for the agent to show up
+        server.wait_until_succeeds("k3s kubectl get node agent")
+
+        for m in machines:
+            m.succeed("k3s check-config")
+
+        server.succeed("k3s kubectl cluster-info")
+        # Also wait for our service account to show up; it takes a sec
+        server.wait_until_succeeds("k3s kubectl get serviceaccount default")
+
+        # Now create a pod on each node via a daemonset and verify they can talk to each other.
+        server.succeed("k3s kubectl apply -f ${networkTestDaemonset}")
+        server.succeed("k3s kubectl rollout status daemonset test")
```

Comment:
As long as we don't set a node selector and don't set any taints on nodes, this schedules a pod on every node. But indeed this only waits for successful rollout of the daemonset, instead of verifying that 3 pods are ready. I don't think this makes a difference here but I will just revert the check so that it does exactly the same as before.